File: java_demo_programs/J5_2021.java
import java.util.Scanner;

public class J5_2021 {
    public static void main(String[] args) {
        int m, n, k;
        int[] rows;
        int[] columns;

        Scanner sc = new Scanner(System.in);
        m = sc.nextInt();
        n = sc.nextInt();
        k = sc.nextInt();
        rows = new int[m];
        columns = new int[n];

        // 0 == Black, 1 == Gold
        for (int i = 0; i < k; i++) {
            String choice = sc.next();
            int index = sc.nextInt() - 1;
            if (choice.equals("R")) {
                rows[index] = rows[index] ^ 1;
            } else {
                columns[index] = columns[index] ^ 1;
            }
        }

        // count
        int count = 0;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                count += rows[i] ^ columns[i];
            }
        }

        System.out.println(count);
    }
}

File: python_demo_programs/J5_2021.py
m = int(input())
n = int(input())
k = int(input())

# 0 == Black, 1 == Gold
rows = [0] * m
columns = [0] * n

for i in range(k):
    choice = input().split()
    if choice[0] == 'R':
        rows[int(choice[1]) - 1] ^= 1
    else:
        columns[int(choice[1]) - 1] ^= 1

# combine and count
count = 0
for i in range(m):
    for j in range(n):
        count += rows[i] ^ columns[j]

print(count)
